File: Section 2/Nicholson_Bailey_Trajectory.m
%% The Mathematics of Host-Parsitoid Population Dynamics:
% This file graphs a time series trajectory of the Nicholson-Bailey Model.
% (This file creates the left panel of Figure 6 in Section 2.2.)
clear all, close all, clc

% Model Parameters: 
c = .1;
R = 2;
k = 1;

% Number of years: 
N = 20; 

% Escape response:
f = @(P) exp(-c*P);

% Initial populations:
H0 = 5;
P0 = 8;

% Vector initialization: 
H = [H0 zeros(1,N-1)];
P = [P0 zeros(1,N-1)];

% Function iteration to create the trajectory:
for t = 1:N
    H(t+1) = R*H(t)*f(P(t));
    P(t+1) = k*R*H(t)*(1-f(P(t)));
end

% Time Series Plot: 
figure(1)
plot(0:N,H,'r-o','linewidth',3)
hold on
plot(0:N,P,'b-o','linewidth',3)
set(gca,'fontsize',18)
title('Trajectory of Nicholson-Bailey Model','fontsize',25,...
      'interpreter','latex')
xlabel('$t$ (years)','fontsize',22,'interpreter','latex')
ylabel('$H_t$, $P_t$','fontsize',22,'interpreter','latex')
legend('Hosts','Parasitoids','interpreter','latex','location','Northwest')
grid on
grid minor

% Phase Plane Plot:
figure(2)
plot(H,P,'k-o','linewidth',3)
hold on
plot(H0,P0,'go','linewidth',5)
set(gca,'fontsize',18)
title('Trajectory of Nicholson-Bailey Model','fontsize',25,...
      'interpreter','latex')
xlabel('$H_t$ (hosts)','fontsize',22,'interpreter','latex')
ylabel('$P_t$ (parsitoids)','fontsize',22,'interpreter','latex')
legend('$(H_t,P_t)$','$(H_0,P_0)$','interpreter','latex','location','Northwest')
grid on
grid minor

File: Python Program Files/Section_2/Nicholson_Bailey_Trajectory.py
"""
%% The Mathematics of Host-Parsitoid Population Dynamics:
% This file graphs a time series trajectory of the Nicholson-Bailey Model.
% (This file creates the left panel of Figure 6 in Section 2.2.)
"""

# Import libraries:
import numpy as np
import matplotlib.pyplot as plt

# Model Parameters: 
c = 0.1
R = 2
k = 1

# Number of years: 
N = 20

# Escape response:
f = lambda P: np.exp(-c * P)

# Initial populations:
H0 = 5
P0 = 8

# Vector initialization: 
H = np.zeros(N + 1)
P = np.zeros(N + 1)
H[0] = H0
P[0] = P0

# Function iteration to create the trajectory:
for t in range(N):
    H[t + 1] = R * H[t] * f(P[t])
    P[t + 1] = k * R * H[t] * (1 - f(P[t]))

# Time Series Plot: 
plt.figure(1)
plt.plot(range(N + 1), H, 'r-o', linewidth=3)
plt.plot(range(N + 1), P, 'b-o', linewidth=3)
plt.gca().tick_params(labelsize=18)
plt.title('Trajectory of Nicholson-Bailey Model', fontsize=25)
plt.xlabel(r'$t$ (years)', fontsize=22)
plt.ylabel(r'$H_t$, $P_t$', fontsize=22)
plt.legend(['Hosts', 'Parasitoids'], loc='upper left')
plt.grid(True, which='both')
plt.minorticks_on()

# Phase Plane Plot:
plt.figure(2)
plt.plot(H, P, 'k-o', linewidth=3)
plt.plot(H0, P0, 'go', markersize=10, markeredgewidth=2)
plt.gca().tick_params(labelsize=18)
plt.title('Trajectory of Nicholson-Bailey Model', fontsize=25)
plt.xlabel(r'$H_t$ (hosts)', fontsize=22)
plt.ylabel(r'$P_t$ (parsitoids)', fontsize=22)
plt.legend([r'$(H_t,P_t)$', r'$(H_0,P_0)$'], loc='upper left')
plt.grid(True, which='both')
plt.minorticks_on()

plt.show()
